sdks/java/extensions/sql/jdbc/src/test/java/org/apache/beam/sdk/extensions/sql/jdbc/BeamSqlLineIT.java

```diff
@@ -0,0 +1,328 @@
+                              + "              FROM taxi_rides\n"
+                              + "              GROUP BY HOP(\n"
+                              + "                 taxi_rides.event_timestamp,\n"
+                              + "                 INTERVAL '1' SECOND, \n"
+                              + "                 INTERVAL '2' SECOND) LIMIT 2");
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // LIMIT 2 is supposed to be parsed into 6 lines.
+    assertTrue(expectedResult.get().size() == 6);
+  }
+
+  /**
+   * Check if topic exists {@param attempts} times. There is 1 sec wait time between two attempts.
+   *
+   * @param attempts number of attempts. Has to be a value > 0
+   */
+  private void checkIfTopicExists(TestPubsub eventsTopic, int attempts)
+      throws InterruptedException, IllegalArgumentException, IOException {
+    if (attempts <= 0) {
+      throw new IllegalArgumentException(
+          String.format("Set %d attempts, which should be > 0", attempts));
+    }
+
+    while (attempts > 0) {
+      List<SubscriptionPath> topics =
+          eventsTopic.listSubscriptions(
+              projectPathFromPath(String.format("projects/%s", project)), eventsTopic.topicPath());
+      if (topics.size() > 0) {
+        return;
+      }
+
+      attempts--;
+      // Sleep 1 sec
+      Thread.sleep(1000);
+    }
+
+    throw new RuntimeException(
+        String.format("Failed to create subscription within %d attempts", attempts));
+  }
+
+  private long convertTimestampToMillis(String timestamp) throws ParseException {
+    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    return dateFormat.parse(timestamp).getTime();
+  }
+
+  private PubsubMessage message(long timestampInMillis, String jsonPayload) {
+    return new PubsubMessage(
+        jsonPayload.getBytes(UTF_8), ImmutableMap.of("ts", String.valueOf(timestampInMillis)));
+  }
+
+  private static class PubsubMessageJSONStringConstructor {
+    private List<String> messageSchema;
+
+    public PubsubMessageJSONStringConstructor(String... schemas) {
+      ImmutableList.Builder<String> builder = ImmutableList.<String>builder();
+      for (String schema : schemas) {
+        builder.add(schema);
+      }
+
+      messageSchema = builder.build();
+    }
+
+    public String construct(Object... values) throws IllegalArgumentException {
+      if (values.length != messageSchema.size()) {
+        throw new IllegalArgumentException(
+            String.format(
+                "length of values %d does not match " + "with size of schema %d",
+                values.length, messageSchema.size()));
+      }
+
+      JSONObject jsonObject = new JSONObject();
```

Comment:
nit: My suggestion would be to use `Jackson` instead of `fastjson`. It is more widely used, and in my opinion is more mature, and has better documentation / examples, plus it is used a lot in few places in Beam codebase already. `fastjson` was introduced  for some reason when SQL `TableProviders` were added, and doesn't seem to be used anywhere else in Beam code. I'd rather not spread it unless there's a reason for it. Probably we should even replace it with Jackson at some point to avoid having to support multiple json parsers in the project.